### ISSUE TITLE
Review 14 Jan 2022 (after porting to Astro)

### ISSUE THREAD

**Issue by b-g** [hfg-gmuend/openmoji.org#151]
**Landing**
<img width="928" alt="Screen Shot 2022-01-14 at 09 09 33" src="https://user-images.githubusercontent.com/480224/149477832-47c801be-50fc-43cc-a902-89cdc5531492.png">

<img width="929" alt="Screen Shot 2022-01-14 at 09 10 19" src="https://user-images.githubusercontent.com/480224/149477896-2b676c24-de6e-4803-8f11-3f4ecd0e3ca6.png">

- Please remove all "share" links in all footers (everwhere on the entire website). Feels to oldschool :)

**Comment by b-g** [comment]
**Library**

<img width="925" alt="Screen Shot 2022-01-14 at 09 10 56" src="https://user-images.githubusercontent.com/480224/149478302-9e33e362-cd4c-4ec5-872b-6202adde096b.png">


<img width="924" alt="Screen Shot 2022-01-14 at 09 16 00" src="https://user-images.githubusercontent.com/480224/149478338-d3a9f33a-7c1b-4b66-8789-7c7f636a3814.png">

- Would be nicer to have a tiny padding between the two rows of dots
- Would not underline every link ... feels too much

**Comment by b-g** [comment]
**Styleguide**

<img width="925" alt="Screen Shot 2022-01-14 at 09 17 06" src="https://user-images.githubusercontent.com/480224/149479401-6c27d61b-8166-4cbb-84a3-ba426a10ed78.png">

- Get started button should be after
- Now all sub-headline on the page a black ... what about giving some color back and maybe also a a section line separator (like in typical READMEs)

<img width="924" alt="Screen Shot 2022-01-14 at 09 18 16" src="https://user-images.githubusercontent.com/480224/149479464-53c393c7-c29e-45fa-91d2-a0d9a30b801f.png">

- I miss the small grey borders

**Comment by b-g** [comment]
**About** 

<img width="922" alt="Screen Shot 2022-01-14 at 09 21 08" src="https://user-images.githubusercontent.com/480224/149481139-8d3d61f4-83b8-4b3f-a641-4e498715d86d.png">

- Having tow time the page title e.g. "About" feels a bit too much. I would remove one and either have only a page title (under the header) or the page title in the header (and no title underneath).

<img width="924" alt="Screen Shot 2022-01-14 at 09 21 26" src="https://user-images.githubusercontent.com/480224/149481207-654b9b5b-a538-4643-b1ff-99dcde235fed.png">

- Missing links

<img width="922" alt="Screen Shot 2022-01-14 at 09 21 58" src="https://user-images.githubusercontent.com/480224/149481219-a38643e6-004f-4146-8229-84b6cdbb50c3.png">

- Padding after headline should be bigger in general

**Comment by b-g** [comment]
Hi @stephanbogner! Many thanks for everything! As discussed ... here a quick review. Hope not too crazy in terms of effort.

**Comment by stephanbogner** [comment]
Hi Bene,

All the points you mentioned should be fixed now – but your pair of 👀  to check would be much appreciated 🙏 

The new version is online on [lab.stephanbogner.de](https://lab.stephanbogner.de/)

### Four aspects worth noting:
**1.: I took the opportunity to also clean up the variable names to be consistent.**
<img width="1083" alt="Screenshot 2022-02-14 at 21 36 02" src="https://user-images.githubusercontent.com/7897006/153942307-d99b026f-91ca-44d9-96e6-124171ca4700.png">

**2.: To get the desired behaviour regarding the buttons (and text) going below the image in the style guide on mobile, I had to create a new component. Because the behaviour is not always desired (in later parts of the styleguide).**
<img width="917" alt="Screenshot 2022-02-14 at 21 30 44" src="https://user-images.githubusercontent.com/7897006/153941857-84ba4a54-ae85-4dcc-9261-d08517be189e.png">

**3.: The "about"-page shares the same page template as the "styleguide"**
= It also features the colored headlines with the line below. Is that alright? (If not, I'll need to create a separate theme)

<img width="1280" alt="Screenshot 2022-02-14 at 21 16 09" src="https://user-images.githubusercontent.com/7897006/153940195-c93045b1-8251-4fb5-9a63-878242cefb9a.png">

**4.: I have set the image border in theme code, so we can use the normal markdown-way to declare images**
= It gets added to all images.

If that is not desired, we would have to either (a) declare a Astro-Component, e.g. `<CustomImg src="xyz.png" alt="ABC" border="true"/>` or (b) use HTML-Syntax in Markdown with a special class, to differentiate in Markdown between the two image styles.

Let me know whether a border around everything is fine or not.

<img width="1280" alt="Screenshot 2022-02-14 at 21 16 25" src="https://user-images.githubusercontent.com/7897006/153940532-01dd8baf-ff94-4143-a01d-0c8e3def95c9.png">

**Comment by b-g** [comment]
@stephanbogner Many thanks for this! 

> Four aspects worth noting
All good + Great!  IMO nothing to do or change.

But found these tiny details ... afterwards happy to launch the site 🥳! :)

Find it odd that the filter / sorting controls scroll down a bit
![2022-02-16 at 16 17](https://user-images.githubusercontent.com/480224/154295881-4cb4c089-bc6d-412b-88f8-ac558888d123.jpg)

Would find it better to have the extras in the sidebar both at the very end like we have it currently 
![2022-02-16 at 16 19](https://user-images.githubusercontent.com/480224/154296198-3923b2fe-313e-43c8-80a2-c8fe8b2b2491.jpg)

Would find it better if the sidebar would stay open after selection e.g. https://openmoji.org/library/#group=activities%2Fevent
![2022-02-16 at 16 21](https://user-images.githubusercontent.com/480224/154296446-53c911f7-a5bf-4a81-b35f-6cdb820f567a.jpg)

**Comment by stephanbogner** [comment]
Hey @b-g 

Thanks for the report!

### RE: Sidebar-sorting
That was almost intentional because I added alphabetic sorting 😄.
I now changed it so, everything is still ordered alphabetically, except if it starts with `extras-`, then it gets put last.

<img width="553" alt="Screenshot 2022-02-27 at 13 48 48" src="https://user-images.githubusercontent.com/7897006/155883086-08b5f413-bd65-435d-969b-bf033ab201a0.png">

### RE: Stickyness scroll
Good call, should be fixed now:
<img width="1025" alt="Screenshot 2022-02-27 at 14 06 08" src="https://user-images.githubusercontent.com/7897006/155883661-e8ab7615-fe98-4d32-841e-7d8b2adc22d1.png">

### RE: Sidebar closing
That's a definitely a bug due to the rewrite, fixed now too.
<img width="642" alt="Screenshot 2022-02-27 at 14 08 43" src="https://user-images.githubusercontent.com/7897006/155883715-7a675709-6b34-4312-9dfc-6381346017b9.png">

---

### The new version is online on [lab.stephanbogner.de](https://lab.stephanbogner.de/)

**Comment by b-g** [comment]
Hi @stephanbogner, great many thanks for improvements / fixes! Solved! Just I get an empty scrollbar now ... please remove. After this ... we should deploy to production 🥳!

![2022-03-04 at 11 53](https://user-images.githubusercontent.com/480224/156750406-a8cf88bb-7f37-4590-b163-e635ac529a71.jpg)

**Comment by stephanbogner** [comment]
I think there is nothing I can do about it 😞

I have to set `overflow-y: scroll`, because otherwise we have the same problem as we have on the current version of the site:
Currently you can't scroll even when the expanded content is larger than the viewport ([example](https://openmoji.org/library/#group=smileys-emotion)). It is just cropped (except you scroll all the way to the bottom of the page).

Afaik it depends on the OS whether to show a scrollbar or not. E.g. on Mac the sidebars are persistently shown (like your screenshot) when a mouse is plugged in.

I could of course bring the old behaviour back, but IMO that is worse than the scrollbar (which is why I changed it).

**Comment by stephanbogner** [comment]
Ok, nevermind. According to the [specs](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-y) `overflow-y: auto` should fix it.

**Comment by b-g** [comment]
👍 + yay! Should we then deploy it?

**Comment by stephanbogner** [comment]
**Sorry for the slow reply, but there was a reason 🙈:**
On Friday I noticed that the generated .html files are unexpectedly big (not only library but also home) ... and I was like 😮 🤔.

The culprit was the openmoji.json, which is 2mb large. This was always the case, but before it was fetched and not directly baked into the html, so at least I never noticed it.

If we were a big tech company we'd have a backend that serves only needed emoji data, but as we aren't 😄 I did the following:
The .json is now gzipped (1.9mb → 116kb) and fetched by the client (so it can also be cached).

The html of /library is still fairly large, but IMO that's a fair tradeoff, because it can be indexed better by search engine and the initial render is much faster.

———
**Long story short:**
I changed something and now I think we can deploy 😅

**Comment by b-g** [comment]
> The .json is now gzipped (1.9mb → 116kb)

Brilliant! Super that you already integrated it into the post install hook!

> I think we can deploy

How? :) 
1. Should I merge the astro branch into master? 
2. Then merge master into production?

Will this do the trick?

**Comment by stephanbogner** [comment]
> How? :)

> Should I merge the astro branch into master?
> Then merge master into production?
> Will this do the trick?

I just pushed the latest build and the adjusted yml-workflow-file, so yes I think this should do the trick 🤞 🙂

**Comment by b-g** [comment]
Thanks + Ok it is running. Fingers crossed :) Closed.
